src/uct/ib/mlx5/ib_mlx5.inl

```diff
@@ -113,12 +113,14 @@ uct_ib_mlx5_inline_copy(void *restrict dest, const void *restrict src, unsigned
 {
     ptrdiff_t n;
 
-    if (dest + length <= wq->qend) {
-        memcpy(dest, src, length);
-    } else {
-        n = wq->qend - dest;
-        memcpy(dest, src, n);
-        memcpy(wq->qstart, src + n, length - n);
+    if (length) {
```

Comment:
looks like extra branch on fast-path? can't we silence the compiler somehow or make it only for ARM?

Reply:
below there are conditions where length used, new condition should not affect to performance.

and original code is not valid: memcpy has unpredictable behavior on NULL ptr in zero-length.

so, look for ARM specific suppression?